CRM/Core/Permission.php

```diff
@@ -611,6 +611,13 @@ protected static function assembleBasicPermissions(): array {
     $allPermissions = array_merge($permissions, $module_permissions);
     // Propagate implied_by permissions to their parents
     foreach ($allPermissions as $name => $permission) {
+      if (!empty($permission['label']) && empty($permission['title'])) {
```

Comment:
Yup, that's the heart of bridging from `label` to `title`.

Reply:
@totten it looks like the test fail is just a test that needs updating - but if I fix that this might be at least worth merging as a step forwards

Reply:
FWIW, when I did the other-way-round, I thought it was kind of tidy to put the bridging code inside the `Hook.php` stub function, as in https://github.com/civicrm/civicrm-core/commit/1e594ccc5652fada3bd0112df966bc059833f301 (except obv that's the opposite). It keeps the conversion encapsulated. This way, the rest of the code in `CRM/Core/Permission.php` (et al) can be converted straight-up. (Search+replace `label`=>`title` in the right spots.)

OTOH, if the idea is to split out to smaller changsets (i.e. there's a period where old+new names are both used by core code), then this kind of placement is probably more forgiving.